cassandra/cqlengine/connection.py

```diff
@@ -82,6 +82,13 @@ def set_session(s):
     session = s
     cluster = s.cluster
 
+    # Set default keyspace from given session's keyspace
+    if not session.keyspace:
+        raise UndefinedKeyspaceException()
```

Comment:
I'm not sure we should blow this up. It's possible to use models without a session keyspace or default with {{__keyspace__}} set on them.

I could see setting models.DEFAULT_KEYSPACE to session.keyspace if default is not set and session is.